torch/_export/__init__.py

```diff
@@ -101,15 +106,22 @@ def _export(
         constraints = []
 
     with torch._dynamo.config.patch(dataclasses.asdict(ExportDynamoConfig())):  # type: ignore[attr-defined]
-        gm, _ = torch._dynamo.export(
-            f,
-            *args,
-            aten_graph=True,
-            tracing_mode="symbolic",
-            decomposition_table=DECOMP_TABLE,
-            constraints=constraints,
-            assume_static_by_default=True,
-        )
+        try:
+            gm, _ = torch._dynamo.export(
+                f,
+                *args,
+                aten_graph=True,
+                tracing_mode="symbolic",
+                decomposition_table=DECOMP_TABLE,
+                constraints=constraints,
+                assume_static_by_default=True,
+            )
+        except (ConstraintViolationError, ValueRangeError) as e:
+            raise UserError(UserErrorType.CONSTRAIN_VIOLATION, str(e))
+        except GuardOnDataDependentSymNode as e:
+            raise UserError(
+                UserErrorType.ANTI_PATTERN,
+                f"Consider annotating your code using constrain_as_*(). {str(e)}")
```

Comment:
Can you please link to an exportdb example? Or do you plan to do it separately?

Reply:
Let me see if we have one for it in the export db

Reply:
It looks like we don't have one yet. Let's do it separately